### PR TITLE
Fix a bunch of clippy warnings

### PR DESCRIPTION
**Pull request by tcharding**
Fix a, hopefully non-controversial, bunch of clippy warnings.

### CODE REVIEW

**Comment by Kixunil** [comment]
Note: it's a clippy bug that it fires since we have MSRV set.

**Comment by Kixunil** [comment]
I've filed an issue: https://github.com/rust-lang/rust-clippy/issues/8276

**Comment by tcharding** [comment]
Are you sure this does fire when MSRV is set? I tried it in an empty crate and it does not and its not there on `master`. I did not comment on your bug report in case I'm wrong.

(My fault, I was working on top of a new edition-2018 branch I did then I pulled out the patches I thought were relevant, I obviously got this one wrong.)

File: src/util/schnorr.rs [hunk 1]
```diff
@@ -220,7 +220,7 @@ impl SchnorrSig {
                 // default type
                 let sig = secp256k1::schnorr::Signature::from_slice(sl)
                     .map_err(SchnorrSigError::Secp256k1)?;
-                return Ok( SchnorrSig { sig, hash_ty : SchnorrSigHashType::Default });
+                Ok( SchnorrSig { sig, hash_ty : SchnorrSigHashType::Default })
```
[the remark this answers is not in the document]
Comment (tcharding) [reply src/util/schnorr.rs:223]: ooo yes, thanks.

[the remark this answers is not in the document]
Comment (tcharding) [reply src/util/schnorr.rs:223]: *cough* rust format.

**Comment by tcharding** [comment]
Thanks for the reviews. I removed the `copied` patch and updated the one with the unwanted extras space (`3088eb1 Remove unneeded calls to return`).

Rebased on master and force pushed.

**Comment by tcharding** [comment]
Fail, fail, fail - closing this and just leaving it until edition-2018 merges.

**Comment by Kixunil** [comment]
@tcharding I did test it and it did fire. Weird.

My `main.rs`:

```rust
fn main() {
    [1, 2, 3].iter().map(|x| *x).collect::<Vec<_>>();
}
```

clippy.toml

```toml
msrv = "1.29"
```

Tested using `cargo +stable clippy`, I have the latest stable via rustup in the VM I tested it in.

**Comment by tcharding** [comment]
I'm fighting with Clippy big time at the moment, my problem is that using `cargo +stable clippy` only shows warnings once (i.e. not the second time you run it), to get around this I use `cargo +nightly -Z unstable-options` but `nightly` does not like the `msrv` field in `clippy.toml` that we use.

**Comment by thomaseizinger** [comment]
> I'm fighting with Clippy big time at the moment, my problem is that using `cargo +stable clippy` only shows warnings once (i.e. not the second time you run it), to get around this I use `cargo +nightly -Z unstable-options` but `nightly` does not like the `msrv` field in `clippy.toml` that we use.

I thought this was fixed a couple of version ago. What version does `stable `refer to on your machine?

**Comment by tcharding** [comment]
1.48 ahhh <ghost face>

**Pull request closed without merging**